services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,603 @@
+     *
+     * @param attributeName Name of the attribute.
+     * @return the value of the specified attribute in the current document as a ByteBuffer; or null if the attribute either
+     * doesn't exist or the attribute value is null.
+     * @throws UnsupportedOperationException If the attribute value involves a byte buffer which is not backed by an accessible
+     *                                       array
+     */
+    ByteBuffer getByteBuffer(String attributeName);
+
+    /**
+     * Gets the Set of String values of the given attribute in the current document.
+     *
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as a set of strings; or null if the attribute either
+     * doesn't exist or the attribute value is null.
+     */
+    Set<String> getStringSet(String attributeName);
+
+    /**
+     * Gets the Set of Number values of the given attribute in the current document.
+     *
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as a set of BigDecimal's; or null if the attribute either
+     * doesn't exist or the attribute value is null.
+     */
+    Set<Number> getNumberSet(String attributeName);
+
+    /**
+     * Gets the Set of Binary values of the given attribute in the current document.
+     *
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as a set of byte arrays; or null if the attribute either
+     * doesn't exist or the attribute value is null.
+     */
+    Set<byte[]> getBinarySet(String attributeName);
+
+    /**
+     * Gets the Set of ByteBuffer values of the given attribute in the current document.
+     *
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as a set of <code>ByteBuffer</code>; or null if the
+     * attribute either doesn't exist or the attribute value is null.
+     */
+    Set<ByteBuffer> getByteBufferSet(String attributeName);
+
+    /**
+     * Gets the List of values of type T for the given attribute in the current document.
+     *
+     * @param attributeName Name of the attribute.
+     * @param <T>           Type T of List elements
+     * @return value of the specified attribute in the current document as a set of <code>ByteBuffer</code>; or null if the
+     * attribute either doesn't exist or the attribute value is null.
+     */
+    <T> List<T> getList(String attributeName);
+
+    /**
+     * Gets the Map  with Key as String and values as type T for the given attribute in the current document.
+     * <p>Note that any numeric type of  map is always canonicalized into {@link SdkNumber}, and therefore if <code>T</code> * 
+     * * referred to a <code>Number</code> type, it would need to be * <code>SdkNumber</code> to avoid a class cast exception.
```

Comment:
Nit: Extra whitespace after Map (207), and before map(208). Extra "*" at end of 208, at beginning of 209, and before <code